src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3127,6 +3127,12 @@ private unsafe void ParseRemaining()
                     _string = _syntax.SchemeName + ":";
                 }
 
+                // If host is absent, uri is abnormal and relative as in RFC 3986 section 5.4.2
+                if (_info.Offset.Host == _info.Offset.Path)
+                {
+                    _string = _syntax.SchemeName + ":";
+                }
+
```

Comment:
I must be missing something... isn't this added if block identical to the one immediately above it?